part-iii-infrastructure-components/query-processing.md

```diff
@@ -12,11 +12,151 @@ The Query Gateway is a convenient interface towards the Query dispatching mechan
+
+(2) If there is a subscription query interested in updates about this specific GiftCard we emit an update. The first parameter of the emission is the type of the query (String in our case) which corresponds to the query type in previously defined query handler. The second parameter is a predicate which will select the subscription query to be updated. In our case we will update only subscription queries interested in the GiftCard which has been updated. The third parameter is the actual update, which in our case is the redeemed amount. There are several overloads of the emit method present, feel free to take a look at JavaDoc for more specifics on that. Important thing to underline here is that an update is a message and that some overloads take the update message as a parameter (in our case we just sent the payload which was wrapped in the message) which enables us to attach meta-data for example. 
+
+Once we have query handling and emitting side implemented, we can issue a subscription query to get initial state of the GiftCard and be updated once this GiftCard is redeemed:
+
+```java
+// (1)
+commandGateway.sendAndWait(new IssueCmd("gc1", 100)); 
+// (2)
+GenericSubscriptionQueryMessage<String, CardSummary, Integer> query =
+                new GenericSubscriptionQueryMessage<>("gc1",
+                                                      ResponseTypes.instanceOf(CardSummary.class),
+                                                      ResponseTypes.instanceOf(Integer.class));
+// (3)
+SubscriptionQueryResult<QueryResponseMessage<CardSummary>, SubscriptionQueryUpdateMessage<Integer>> queryResult =
+                queryBus.subscriptionQuery(query);
+// (4)
+queryResult.handle(System.out::println, System.out::println);
+// (5)	
+commandGateway.sendAndWait(new RedeemCmd("gc1", 10));			
+```
+
+(1) Issuing a GiftCard with "gc1" id and initial value of 100.
+
+(2) Creating a subscription query message to get the initial state of "gc1" GiftCard (this initial state is of type `CardSummary`) and to be updated once the state of GiftCard with id "gc1" is changed (in our case update means the card is redeemed). The type of the update is an `Integer`. Do note that the type of the update must match the type of the emission side.
+
+(3) Once the message is created, we are sending it via the `QueryBus`. We receive a query result which contains two components: one is `initialResult` and the other is `updates`. In order to achieve 'reactiveness' we use [Project Reactor](https://projectreactor.io/)'s `Mono` for `initialResult` and `Flux` for `updates`. 
+
+> **Note**
+> Once the subscription query is issued, all updates are queued until the subscription to the `Flux` of `updates` is done. This behavior prevents losing of updates.
+
+> **Note**
+> `reactor-core` dependency is mandatory for usage of subscription queries. However, it is a compile time dependency and it is not required for other Axon features.
+
```

Comment:
Maybe you can specify that you cannot request more than one subscription with the same subscription identifier.